src/gui/accountsettings.cpp

```diff
@@ -638,7 +638,11 @@ void AccountSettings::slotAccountStateChanged()
                                  Utility::escape(safeUrl.toString()));
         QString serverWithUser = server;
         if (AbstractCredentials *cred = account->credentials()) {
-            serverWithUser = tr("%1 as <i>%2</i>").arg(server, Utility::escape(cred->user()));
+            QString user = account->davDisplayName();
```

Comment:
You dont need to instantiate new string, just do ```if(account->davDisplayName().isEmpty())``` and then e.g. ```serverWithUser = tr("%1 as <i>%2</i>").arg(server, Utility::escape(account->davDisplayName()));```

Reply:
But whatever seems clearer to you is good.

Reply:
QString is implicitly shared. This will just takes stores the pointer on the stack inside that `QString` until the end of the block and won't impact memory usage at all.

Repeating `davDisplayName()` would actually be slightly worse since the function isn't inlined and you'd call it twice across DLLs, but performance-wise this also won't make any noticeable difference unless you call this 1M times either anyway.

Reply:
Qt will not stop suprise me, good to know